library/core/src/time.rs

```diff
@@ -29,6 +29,19 @@ const NANOS_PER_MICRO: u32 = 1_000;
 const MILLIS_PER_SEC: u64 = 1_000;
 const MICROS_PER_SEC: u64 = 1_000_000;
 
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[rustc_layout_scalar_valid_range_start(0)]
+#[rustc_layout_scalar_valid_range_end(999_999_999)]
+struct Nanoseconds(u32);
```

Comment:
Not sure if it makes any difference, but I usually prefer newtypes around integers and floats to be `#[repr(transparent)]` to ensure you get their ABI instead of struct ABI. This mostly only makes a difference on Windows though afaik.

Reply:
@CryZe Not sure if it makes a difference either, but that seems reasonable to me.